src/components/ButtonWithProgress.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { FC } from 'react';
+import { Button } from '@material-ui/core';
+import Progress from './Progress';
+import styled from 'styled-components';
+
+const StyledProgressContainer = styled.div`
+  padding-left: 1rem;
+  display: flex;
+  align-items: center;
+`;
+
+interface ButtonWithProgressProps {
+  buttonText: string;
+  isLoading: boolean;
+  onClick?: () => void;
+  type?: 'submit' | 'reset' | 'button';
+}
+
+const ButtonWithProgress: FC<ButtonWithProgressProps> = ({ buttonText, isLoading, onClick, type }) => {
```

Comment:
type='button'?

Reply:
mulig det funker like fint med undefined også (hvis man ikke angir type)... idk

Reply:
jeg har aldri brukt type='button' tidligere, så sikkert fint som det er!

Reply:
det er de valgene man får for type propen i MUIButton. De har ikke en egen type for det, bare listet som strenger sånn der. trenger ikke sette default prop, MUI setter det til type="button" hvis det ikke spesifiseres